src/IntervalSets.jl

```diff
@@ -229,6 +229,13 @@ Clamp the scalar `t` such that the result is in the interval `i`.
 clamp(t, i::TypedEndpointsInterval{:closed,:closed}) =
     clamp(t, leftendpoint(i), rightendpoint(i))
 
+"""
+    mod(x, i::ClosedInterval)
+
+Find `y` in the `i` interval such that `x ≡ y (mod n)`, where `n = width(i)`.
+"""
+mod(x, i::TypedEndpointsInterval{:closed,:closed}) = mod(x - leftendpoint(i), width(i)) + leftendpoint(i)
```

Comment:
Do we need this method for `TypedEndpointsInterval{:closed,:closed}`? I thought the method should be defined for `TypedEndpointsInterval{:open,:closed}` and `TypedEndpointsInterval{:closed,:open}` because:

* `mod(3, 1..3) != 3` seems confusing.
* ℝ = ⋯ ⊔ (-2,-1] ⊔ (-1,0] ⊔ (0,1] ⊔ (1,2] ⊔ ⋯ holds, but ℝ = ⋯ ⊔ [-2,-1] ⊔ [-1,0] ⊔ [0,1] ⊔ [1,2] ⊔ ⋯ is not correct.

I think it might be okay to have the method for `TypedEndpointsInterval{:closed,:closed}`, but the docstring can be improved with the prior example in `jldoctest`.

Reply:
I agree with that. Do you intend to implement the methods for half-open intervals in PR? I'm still thinking we would like to have the methods for mathematical correctness. But this can be done in another PR, so we may merge as is or wait for the changes.